### PR TITLE
Replace SpanData.hasRemoteParent/getParentSpanId with getParentSpanContext

### LINKED ISSUE #987: Remove SpanData.getHasRemoteParent() and the logic to determine that
This property was removed from the proto definition and is no longer a requirement in OpenTelemetery.

### PR DESCRIPTION
**Pull request by anuraaga**
Fixes #987 

~~From what I can tell, @Oberon00's concern is addressed by now since we have `Context` everywhere.~~ added getParentSpanContext to keep the existing functionality while also being able to remove more methods.

~~Since this is export-side, I just deleted it but let me know if it's worth doing `@Deprecated` first.~~

### CODE REVIEW

**Review by Oberon00 — CHANGES_REQUESTED** [review]
I think this feature is useful.

**Comment by anuraaga** [comment]
@Oberon00 Went ahead and added `getParentSpanContext`

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/data/SpanData.java [hunk 1]
```diff
@@ -50,13 +50,36 @@
...
+
   /**
    * Returns the parent {@code SpanId}. If the {@code Span} is a root {@code Span}, the SpanId
    * returned will be invalid..
    *
    * @return the parent {@code SpanId} or an invalid SpanId if this is a root {@code Span}.
+   * @deprecated Use {@link #getParentSpanContext()}
    */
-  String getParentSpanId();
+  @Deprecated
```
Comment (Oberon00) [inline sdk/trace/src/main/java/io/opentelemetry/sdk/trace/data/SpanData.java:66]: I don't know if we even want to deprecate that. So far, it seems exporters have only needed this, so it seems OK to keep it as convenience shortcut.
I'm fine with deprecating/removing it though, just a thought.

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java [hunk 2]
```diff
@@ -135,10 +131,8 @@ private RecordEventsReadableSpan(
    * @param context supplies the trace_id and span_id for the newly started span.
    * @param name the displayed name for the new span.
    * @param kind the span kind.
-   * @param parentSpanId the span_id of the parent span, or {@code Span.INVALID} if the new span is
-   *     a root span.
-   * @param hasRemoteParent {@code true} if the parentContext is remote. {@code false} if this is a
-   *     root span.
+   * @param parentSpanContext the span_id of the parent span, or {@code Span.INVALID} if the new
```
Comment (Oberon00) [inline sdk/trace/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java:134]:
```suggestion
   * @param parentSpanContext the parent span context, or {@link SpanContext#getInvalid()} if this
```

**Review by Oberon00 — APPROVED** [review]

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/data/SpanData.java (hunk 1 above)
Comment (jkwatson) [reply sdk/trace/src/main/java/io/opentelemetry/sdk/trace/data/SpanData.java:66]: I'm fine either way.

**Review by jkwatson — APPROVED** [review]
Thanks!

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/data/SpanData.java (hunk 1 above)
Comment (anuraaga) [reply sdk/trace/src/main/java/io/opentelemetry/sdk/trace/data/SpanData.java]: Makes sense to me to keep it so undeprecated

**Pull request merged**